src/parser/type_checking.c

```diff
@@ -6089,6 +6089,8 @@ pt_apply_expressions_definition (PARSER_CONTEXT * parser, PT_NODE ** node)
   int matches = 0, best_match = -1, i = 0;
   EXPRESSION_SIGNATURE sig;
 
+  static bool oracle_style_divide = prm_get_bool_value (PRM_ID_ORACLE_STYLE_DIVIDE);
```

Comment:
How about just changing below and setting 'oracle_style_number_return=y"?

```
diff --git a/src/parser/type_checking.c b/src/parser/type_checking.c
index d27a1891b..791d76fc4 100644
--- a/src/parser/type_checking.c
+++ b/src/parser/type_checking.c
@@ -10580,7 +10580,11 @@ pt_common_type (PT_TYPE_ENUM arg1_type, PT_TYPE_ENUM arg2_type)
          /* The common type between two ENUMs is string */
          common_type = PT_TYPE_VARCHAR;
        }
-      else
+      else if (PT_IS_DISCRETE_NUMBER_TYPE (arg1_type) && PT_IS_DISCRETE_NUMBER_TYPE (arg1_type))
+        {
+         common_type = PT_TYPE_NUMERIC;
+       }
+      else
        {
          common_type = arg1_type;
        }
```

```
create table t1 (s_c1 short, s_c2 short, i_c1 int, i_c2 int, b_c1 bigint, b_c2 bigint);
insert into t1 values (1, 2, 1, 2, 1, 2);

select s_c1/s_c2, i_c1/i_c2, b_c1/b_c2, 1/2, 1.0/2 from t1;
/*
  s_c1/s_c2             i_c1/i_c2             b_c1/b_c2             1/2                   1.0/2
==============================================================================================================
  0.5                   0.5                   0.5                   0.5                   0.5
*/

select s_c1+s_c2, i_c1+i_c2, b_c1+b_c2, 1+2, 1.0+2 from t1;
/*
  s_c1+s_c2             i_c1+i_c2             b_c1+b_c2             1+2                   1.0+2
==============================================================================================================
  3                     3                     3                     3                     3
*/

select s_c1-s_c2, i_c1-i_c2, b_c1-b_c2, 1-2, 1.0-2 from t1;
/*
  s_c1-s_c2             i_c1-i_c2             b_c1-b_c2             1-2                   1.0-2
==============================================================================================================
  -1                    -1                    -1                    -1                    -1
*/

select s_c1*s_c2, i_c1*i_c2, b_c1*b_c2, 1*2, 1.0*2 from t1;
/*
  s_c1*s_c2             i_c1*i_c2             b_c1*b_c2             1*2                   1.0*2
==============================================================================================================
  2                     2                     2                     2                     2
*/

Reply:
Even in this case, there are parts that require additional code modifications.

```
select 0.00000001/2, typeof (0.00000001/2),  0.0000000001/2, typeof (0.0000000001/2), 0.00000000000000000001/2, typeof (0.00000000000000000001/2);
/*
  0.00000001/2           typeof(0.00000001/2)  0.0000000001/2         typeof(0.0000000001/2)  0.00000000000000000001/2   typeof(0.00000000000000000001/2)
====================================================================================================================================
  0.000000005           'numeric (9, 9)'      0.0000000001          'numeric (10, 10)'    0.00000000000000000001  'numeric (20, 20)'
*/

Reply:
The NUMERIC type has precision and scale info which might not be defined at compile time. 
SELECT count(*) / 3 FROM t1 query returns integer.

Reply:
It should not be changed in the pt_common_type function, it should be changed in the pt_common_type_op function so that the operator can be checked and changed.
```
diff --git a/src/parser/type_checking.c b/src/parser/type_checking.c
index d27a1891b..cf40e6816 100644
--- a/src/parser/type_checking.c
+++ b/src/parser/type_checking.c
@@ -11310,6 +11310,12 @@ pt_common_type_op (PT_TYPE_ENUM t1, PT_OP_TYPE op, PT_TYPE_ENUM t2)

   switch (op)
     {
+    case PT_DIVIDE:
+      if (PT_IS_DISCRETE_NUMBER_TYPE (t1) && PT_IS_DISCRETE_NUMBER_TYPE (t2))
+        {
+         result_type = PT_TYPE_NUMERIC;
+       }
+       break;
     case PT_MINUS:

```